### PR TITLE
Cleanup and refactoring

### LINKED ISSUE #57: Codebase re-factoring / cleanup issue
In order to be as friendly to developers as possible, we should take what we have in 0.1 and make sure the code is organized well and re-factored so that all concerns are as separate and clear as possible.

### PR DESCRIPTION
**Pull request by bgins**
# Description

This PR cleans up and refactors the codebase with the following changes:

- [x] Move to production by removing staging endpoints
- [x] Re-organize imports to order them consistently across modules
- [x] Break `webnative.ts` up into `init.ts` and `auth/account.ts` modules
- [x] Move the `utils.ts` module from `src/lib/common` to `src/lib`
- [x] Replace the `Toast` used in `__layout.svelte` with a `Notification` and remove the `Toast` component
- [x] Move the theme implementation from `src/lib/theme` to  `src/lib` (no need for the subdirectory here)
- [x] Rename `theme` to `themeStore` to name it
- [ ] ~~Add missing dark borders~~ (Seems this is all covered!)
- [x] Enforce minimum node version (at `>=16.9` which is minimum for SvelteKit) 
- [x] Check that all subscriptions are unsubscribed on destroy
- [x] Examine components for code that can be lifted to their parent route. In particular code that is re-used by sibling components, but some functionality like URL parsing might make more sense in a route
- [x] Replace `uuid` helper in `utils.ts` with [WebCrypto `randomUUID`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)
- [x] Remove unneeded console logs
- [x] Generalize URL parsing code
- [x] Hide app name in header on mobile
- [x] Remove `deviceStore` (It was unused. We are setting mobile styles with Tailwind utility classes.)
- [x] Remove unused navigate event from `AreYouSure` component
- [x] Add social preview image and add URL to `app-info.ts`
- [x] Replace `convertUint8ToString` in `utils.ts` with [`toString` from uint8arrays](https://www.npmjs.com/package/uint8arrays#tostringarray-encoding--utf8)
- [x] Separate gallery directory creation into dedicated function
- [x] Update `webnative` version in `package.json` to allow minor upgrades

Under consideration:

- ~~Create gallery directories in gallery route instead of at account registration~~
- ~~Upgrade SvelteKit to latest beta version~~ The changes are significant and should be on a separate PR.
- ~~Add icons for info and warning notifications~~ We can add these later when we have the assets available

## Link to issue

Closes #57 

## Type of change

Please delete options that are not relevant.

- [x] Refactor (non-breaking change that updates existing functionality)

### CODE REVIEW

**Comment by bgins** [comment]
@avivash requesting your review because there a couple of changes to the gallery in this PR. Most notably:

- Replace `uuid` helper in `utils.ts` with [WebCrypto `randomUUID`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)
- Replace `convertUint8ToString` in `utils.ts` with [`toString` from uint8arrays](https://www.npmjs.com/package/uint8arrays#tostringarray-encoding--utf8)

Don't feel obligated to review this entire PR (unless you want to!). Wanted to run the gallery changes by you though if you have a chance to look at them.

File: src/lib/notifications.ts [hunk 1]
```diff
@@ -22,7 +21,7 @@ export const addNotification: (
   timeout?: number
 ) => void = (msg, type = 'info', timeout = 5000) => {
   // uuid for each notification
-  const id = uuid()
+  const id = crypto.randomUUID()
```
Comment (avivash) [inline src/lib/notifications.ts:24]: Nice 👌🏼

**Comment by avivash** [comment]
> @avivash requesting your review because there a couple of changes to the gallery in this PR. Most notably:
> 
> * Replace `uuid` helper in `utils.ts` with [WebCrypto `randomUUID`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)
> * Replace `convertUint8ToString` in `utils.ts` with [`toString` from uint8arrays](https://www.npmjs.com/package/uint8arrays#tostringarray-encoding--utf8)
> 
> Don't feel obligated to review this entire PR (unless you want to!). Wanted to run the gallery changes by you though if you have a chance to look at them.

That sounds good! I'm definitely a fan of using the native `crypto` package(not the npm one) wherever possible 👌🏼. I like the `toString` from `uint8arrays` more too. I hadn't really had to use `uint8Arrays` much before I started working with webnative, so I had a feeling I may not be using best practises 😅 thanks for updating that

It may be worth mentioning I ended up building out the gallery in walletauth using a DDD approach to make it easier for people to remove all the gallery code in one fell swoop if they decide they don't want it https://github.com/webnative-examples/walletauth/tree/main/src/routes/gallery. I'd be happy to look into implementing something similar in this repo 👍🏼 

Also, I don't think this is related to your app, but when I ran your branch locally I noticed the homepage was showing my eth address as my username(even though I don't have walletauth or any other node apps currently running). This does default to the same port as the walletauth app when running `dev` though, so maybe I'll change the default port in the walletauth app for now? 
image

**Comment by bgins** [comment]
> That sounds good! I'm definitely a fan of using the native `crypto` package(not the npm one) wherever possible 👌🏼. I like the `toString` from `uint8arrays` more too. I hadn't really had to use `uint8Arrays` much before I started working with webnative, so I had a feeling I may not be using best practises 😅 thanks for updating that

Thanks for looking those over! I might be hesitant to use `uint8Arrays` for a small thing like this, but we use it heavily in Webnative so no harm using it here as well. 🙂 

> It may be worth mentioning I ended up building out the gallery in walletauth using a DDD approach to make it easier for people to remove all the gallery code in one fell swoop if they decide they don't want it https://github.com/webnative-examples/walletauth/tree/main/src/routes/gallery. I'd be happy to look into implementing something similar in this repo 👍🏼

Yeah, making it easier to pull out would be great! I'd say let's wait until we get upgraded to the latest SvelteKit. Looks that will bring things closer to the walletauth implementation and easier to port over.

> Also, I don't think this is related to your changes, but when I ran your branch locally I noticed the homepage was showing my eth address as my username(even though I don't have walletauth or any other node apps currently running). This does default to the same port as the walletauth app when running `dev` though, so maybe I'll change the default port in the walletauth app for now? image

Yep, the username and filesystem are stored by domain and port. Switching the default port over there makes sense -- could get confusing if you are working on both projects at the same time.

**Review by jessmartin — APPROVED** [review]
LGTM! 👏🎉

**Pull request merged**